src/backend/utils/misc/guc.c

```diff
@@ -12216,7 +12216,36 @@ ProcessGUCArray(ArrayType *array, GucSource source)
 		 * GPSQL needs to dispatch the database/user config to segments.
 		 */
 		if (Gp_role == GP_ROLE_DISPATCH)
-			appendStringInfo(&MyProcPort->override_options, "-c %s=%s ", name, value);
+		{
+			unsigned int	 j, start, size;
+			char			*temp;
+
+			size = 256;
+			temp = guc_malloc(ERROR, size + 8);
+
+			j = 0;
+			for (start = 0; start < strlen(value); ++start)
+			{
+				if (j == size)
+				{
```

Comment:
I think this part of code has kind of duplication with the code segment in executormgr.c?? Shall we extract this out?